public/wp-content/themes/glific/functions.php

```diff
@@ -37,3 +37,28 @@ function glific_theme_setup(){
 	}
 	add_action('init','glific_theme_setup');
 }
+
+function get_thumbnail_from_youtube_video($url) {
+	$shortUrlRegex = '/youtu.be\/([a-zA-Z0-9_]+)\??/i';
+	$longUrlRegex = '/youtube.com\/((?:embed)|(?:watch))((?:\?v\=)|(?:\/))(\w+)/i';
+	if (preg_match($longUrlRegex, $url, $matches)) {
+		$youtube_id = $matches[count($matches) - 1];
+	}
+	if (preg_match($shortUrlRegex, $url, $matches)) {
+		$youtube_id = $matches[count($matches) - 1];
+	}
+
+	return 'https://img.youtube.com/vi/'.$youtube_id.'/0.jpg';
```

Comment:
What if both preg_match() doesn't match anything?

Reply:
It is checking either embeded url or the youtube ure, will returent the result accordingly